crates/fuel-core/src/schema/chain.rs

```diff
@@ -83,22 +77,15 @@ impl ConsensusParameters {
 #[Object]
 impl ChainInfo {
     async fn name(&self, ctx: &Context<'_>) -> async_graphql::Result<String> {
-        let db = ctx.data_unchecked::<Database>().clone();
-        let name = db
-            .get_chain_name()?
-            .unwrap_or_else(|| DEFAULT_NAME.to_string());
-        Ok(name)
+        let data = ChainQueryContext(ctx.data_unchecked());
+        Ok(data.name()?)
     }
 
     async fn latest_block(&self, ctx: &Context<'_>) -> async_graphql::Result<Block> {
-        let db = ctx.data_unchecked::<Database>().clone();
-        let height = db.get_block_height()?.unwrap_or_default();
-        let id = db.get_block_id(height)?.unwrap_or_default();
-        let block = db
-            .storage::<FuelBlocks>()
-            .get(&id)?
-            .ok_or(not_found!(FuelBlocks))?;
-        Ok(Block::from(block.into_owned()))
+        let query = BlockQueryContext(ctx.data_unchecked());
+
+        let latest_block = query.latest_block()?.into();
+        Ok(latest_block)
     }
 
     async fn base_chain_height(&self) -> U64 {
```

Comment:
just realized we should probably update this to fetch da_height from the relayer